From c4caec74bda9eefbde8fd8015706667c9e1d268f Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <cgarcia@igalia.com>
Date: Thu, 7 Nov 2024 04:31:03 -0800
Subject: [PATCH] REGRESSION(285310@main): [GTK] Cannot open links on various
 websites https://bugs.webkit.org/show_bug.cgi?id=282409

Reviewed by Adrian Perez de Castro.

Since 285310@main the new web views created for a related view on
WebKit::create signal will not share the web process if noopener is
present. So, we need to remove the check that ensured that.

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewCreateNewPage):

Canonical link: https://commits.webkit.org/286275@main
---
 Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp b/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
index f77502d40e7e3..77b60a144e8d7 100644
--- a/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
+++ b/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
@@ -2659,8 +2659,6 @@ void webkitWebViewSetIcon(WebKitWebView* webView, const LinkIcon& icon, API::Dat
 
 RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, Ref<API::PageConfiguration>&& configuration, WebKitNavigationAction* navigationAction)
 {
-    auto& openerInfo = configuration->openerInfo();
-
     ASSERT(!webView->priv->configurationForNextRelatedView);
     SetForScope configurationScope(webView->priv->configurationForNextRelatedView, WTFMove(configuration));
 
@@ -2670,14 +2668,8 @@ RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, Ref<API:
         return nullptr;
 
     Ref newPage = getPage(newWebView);
-    if (&getPage(webView) != newPage->configuration().relatedPage() || openerInfo != newPage->configuration().openerInfo()) {
-        g_warning("WebKitWebView returned by WebKitWebView::create signal was not created with the related WebKitWebView");
-        return nullptr;
-    }
-
     ASSERT(newPage->configuration().windowFeatures());
     webkitWindowPropertiesUpdateFromWebWindowFeatures(newWebView->priv->windowProperties.get(), *newPage->configuration().windowFeatures());
-
     return newPage;
 }
 
